server/s3_copy_object_action.cc

```diff
@@ -513,3 +515,27 @@ bool S3CopyObjectAction::if_source_and_destination_same() {
   return ((source_bucket_name == request->get_bucket_name()) &&
           (source_object_name == request->get_object_name()));
 }
+
+void S3CopyObjectAction::set_authorization_meta() {
+  s3_log(S3_LOG_DEBUG, request_id, "Entering\n");
+  auth_client->set_acl_and_policy(bucket_metadata->get_encoded_bucket_acl(),
+                                  bucket_metadata->get_policy_as_json());
+  request->set_action_str("PutObject");
+  next();
+  s3_log(S3_LOG_DEBUG, "", "Exiting\n");
+}
+
+void S3CopyObjectAction::set_source_bucket_authorization_metadata() {
+  s3_log(S3_LOG_INFO, request_id, "Entering\n");
+  auth_client->set_get_method = true;
+  std::string uri = "/" + source_bucket_name + "/" + source_object_name;
+  auth_client->clientabsoulte_uri = uri;
+  auth_client->set_acl_and_policy(
+      source_object_metadata->get_encoded_object_acl(),
+      source_bucket_metadata->get_policy_as_json());
+  request->set_action_str("GetObject");
+  auth_client->check_authorization(
```

Comment:
can we separate this, set function should only set values and authorization need to happen in different function